### ISSUE TITLE
Bug Report At MySpringDataRedisBasedProxyManager

### ISSUE THREAD

**Issue by MisterChangRay** [bucket4j/bucket4j#317]
Environment

java8&springboot-2.3.2.RELEASE&bucket4j&springboot-redis;

Description

When I Use the `buckt.tryConsume(1);`,  i got this exception:

in funciton `compareAndSwap` has four luna script used; 

only `this.scriptCompareAndSwapPx` & this.scriptCompareAndSwapPx` has define return value(0 or 1)  in lua script

but `this.scriptSetNxPx & this.scriptSetNx` not yet return value

so spring-boot-redis return value should be : 'OK' or 'Null',  is byte array, not a number;

in the java code on `compareAndSwap()` , all script result be force cast to Boolean,

so when we use  `scriptSetNxPx ` fuction, we  get that exception; 

[redis set command](https://redis.io/commands/set/)

![image](https://user-images.githubusercontent.com/16421384/196914838-8349b39a-bd8d-41c5-a389-5e29b7622253.png)

**Comment by vladimir-bukhtoyarov** [comment]
@MisterChangRay hello

Please provide stacktrace of Exception if possible.

**Comment by vladimir-bukhtoyarov** [comment]
@MisterChangRay 

Current implementation of SpringDataRedisBasedProxyManager was tested with ```org.springframework.data:spring-data-redis:2.6.4``` .  You use ```springboot-2.3.2``` that depends from older version ```org.springframework.data:spring-data-redis:2.3.2.RELEASE``` https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-data-redis/2.3.2.RELEASE

Looks that your PR works well for both version. I will merge it(and release new version) at weekend.

**Comment by MisterChangRay** [comment]
Caused by: java.lang.ClassCastException: class [B cannot be cast to class java.lang.Long ([B and java.lang.Long are in module java.base of loader 'bootstrap')
	at org.springframework.data.redis.connection.jedis.JedisScriptReturnConverter.convert(JedisScriptReturnConverter.java:53) ~[spring-data-redis-2.3.2.RELEASE.jar:2.3.2.RELEASE]
	at org.springframework.data.redis.connection.jedis.JedisScriptingCommands.eval(JedisScriptingCommands.java:129) ~[spring-data-redis-2.3.2.RELEASE.jar:2.3.2.RELEASE]
	at org.springframework.data.redis.connection.DefaultedRedisConnection.eval(DefaultedRedisConnection.java:1495) ~[spring-data-redis-2.3.2.RELEASE.jar:2.3.2.RELEASE]
	at io.github.bucket4j.redis.spring.cas.SpringDataRedisBasedProxyManager.compareAndSwap(SpringDataRedisBasedProxyManager.java:128) ~[bucket4j_jdk8-redis-8.1.0.jar:?]
	at io.github.bucket4j.redis.spring.cas.SpringDataRedisBasedProxyManager.access$400(SpringDataRedisBasedProxyManager.java:41) ~[bucket4j_jdk8-redis-8.1.0.jar:?]
	at io.github.bucket4j.redis.spring.cas.SpringDataRedisBasedProxyManager$1.compareAndSwap(SpringDataRedisBasedProxyManager.java:80) ~[bucket4j_jdk8-redis-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.generic.compare_and_swap.AbstractCompareAndSwapBasedProxyManager.execute(AbstractCompareAndSwapBasedProxyManager.java:79) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.generic.compare_and_swap.AbstractCompareAndSwapBasedProxyManager.execute(AbstractCompareAndSwapBasedProxyManager.java:51) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.AbstractProxyManager$DefaultRemoteBucketBuilder$1.execute(AbstractProxyManager.java:197) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.DefaultBucketProxy.execute(DefaultBucketProxy.java:222) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.DefaultBucketProxy.tryConsumeImpl(DefaultBucketProxy.java:83) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.AbstractBucket.tryConsume(AbstractBucket.java:201) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at com.gm.open.gateway.filter.RequestLimiterFilter.filter(RequestLimiterFilter.java:134) ~[classes/:?]

**Comment by MisterChangRay** [comment]
> @MisterChangRay hello
> 
> Please provide stacktrace of Exception if possible.

```java
Caused by: java.lang.ClassCastException: class [B cannot be cast to class java.lang.Long ([B and java.lang.Long are in module java.base of loader 'bootstrap')
	at org.springframework.data.redis.connection.jedis.JedisScriptReturnConverter.convert(JedisScriptReturnConverter.java:53) ~[spring-data-redis-2.3.2.RELEASE.jar:2.3.2.RELEASE]
	at org.springframework.data.redis.connection.jedis.JedisScriptingCommands.eval(JedisScriptingCommands.java:129) ~[spring-data-redis-2.3.2.RELEASE.jar:2.3.2.RELEASE]
	at org.springframework.data.redis.connection.DefaultedRedisConnection.eval(DefaultedRedisConnection.java:1495) ~[spring-data-redis-2.3.2.RELEASE.jar:2.3.2.RELEASE]
	at io.github.bucket4j.redis.spring.cas.SpringDataRedisBasedProxyManager.compareAndSwap(SpringDataRedisBasedProxyManager.java:128) ~[bucket4j_jdk8-redis-8.1.0.jar:?]
	at io.github.bucket4j.redis.spring.cas.SpringDataRedisBasedProxyManager.access$400(SpringDataRedisBasedProxyManager.java:41) ~[bucket4j_jdk8-redis-8.1.0.jar:?]
	at io.github.bucket4j.redis.spring.cas.SpringDataRedisBasedProxyManager$1.compareAndSwap(SpringDataRedisBasedProxyManager.java:80) ~[bucket4j_jdk8-redis-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.generic.compare_and_swap.AbstractCompareAndSwapBasedProxyManager.execute(AbstractCompareAndSwapBasedProxyManager.java:79) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.generic.compare_and_swap.AbstractCompareAndSwapBasedProxyManager.execute(AbstractCompareAndSwapBasedProxyManager.java:51) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.AbstractProxyManager$DefaultRemoteBucketBuilder$1.execute(AbstractProxyManager.java:197) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.DefaultBucketProxy.execute(DefaultBucketProxy.java:222) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.distributed.proxy.DefaultBucketProxy.tryConsumeImpl(DefaultBucketProxy.java:83) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at io.github.bucket4j.AbstractBucket.tryConsume(AbstractBucket.java:201) ~[bucket4j_jdk8-core-8.1.0.jar:?]
	at com.gm.open.gateway.filter.RequestLimiterFilter.filter(RequestLimiterFilter.java:134) ~[classes/:?]
```

**Comment by MisterChangRay** [comment]
> 

Thanks a lot

**Comment by MisterChangRay** [comment]
> @MisterChangRay
> 
> Current implementation of SpringDataRedisBasedProxyManager was tested with `org.springframework.data:spring-data-redis:2.6.4` . You use `springboot-2.3.2` that depends from older version `org.springframework.data:spring-data-redis:2.3.2.RELEASE` https://mvnrepository.com/artifact/org.springframework.boot/spring-boot-starter-data-redis/2.3.2.RELEASE
> 
> Looks that your PR works well for both version. I will merge it(and release new version) at weekend.

Thanks a lot

**Comment by vladimir-bukhtoyarov** [comment]
Fix has been released with version 8.1.1
